presto-kudu/src/main/java/io/prestosql/plugin/kudu/KuduSplitManager.java

```diff
@@ -59,4 +79,56 @@ public ConnectorSplitSource getSplits(
                 throw new IllegalArgumentException("Unknown splitSchedulingStrategy: " + splitSchedulingStrategy);
         }
     }
+
+    private static CompletableFuture<?> whenCompleted(DynamicFilter dynamicFilter)
+    {
+        if (dynamicFilter.isAwaitable()) {
+            return dynamicFilter.isBlocked().thenCompose(o -> whenCompleted(dynamicFilter));
+        }
+        else {
+            return DynamicFilter.NOT_BLOCKED;
+        }
+    }
+
+    private static class KuduDynamicFilteringSplitSource
+            implements ConnectorSplitSource
+    {
+        private final CompletableFuture<ConnectorSplitSource> future;
+
+        private KuduDynamicFilteringSplitSource(CompletableFuture<ConnectorSplitSource> future)
+        {
+            this.future = future;
```

Comment:
`requireNotNull(future, "future is null")`

Reply:
updated